docker/docker_client.go

```diff
@@ -627,9 +636,11 @@ func (c *dockerClient) getExtensionsSignatures(ctx context.Context, ref dockerRe
 		return nil, err
 	}
 	defer res.Body.Close()
-	if res.StatusCode != http.StatusOK {
-		return nil, errors.Wrapf(client.HandleErrorResponse(res), "Error downloading signatures for %s in %s", manifestDigest, ref.ref.Name())
```

Comment:
Dropping `client.HandleErrorResponse(res)` seems rather undesirable.

Reply:
Could you elaborate why? I was found the output of `HandleErrorResponse` not very user friendly and hard to brain parse. Is there some behavior or output you'd like to keep?

Reply:
In case, I'd switch all checks to that to have ~consistent error messaging.

Reply:
- It’s the only way to get actual server-produced error messages; dumping the raw HTTP response body would be much worse. Sure, they are often useless, but at least they _can_ be improved.
- The failures are machine-identifiable and converted to specific Go types. See how many different failures in  See `vendor/github.com/docker/distribution/registry/api/v2/errors.go` just map to `http.StatusBadRequest`, and `isManifestInvalidError`.

Reply:
> In case, I'd switch all checks to that to have ~consistent error messaging.

Sure, as long as the API documents that this format of error reporting is used. That’s not the case for look-aside signatures, at the very least; I haven’t checked the other cases.

Reply:
> In case, I'd switch all checks to that to have ~consistent error messaging.

That won't work for all responses:
```
error parsing HTTP 429 response body: invalid character '<' looking for beginning of value: "<html>\r\n<head><title>429 Too Many Requests</title></head>\r\n<body>\r\n<center><h1>429 Too Many Requests</h1></center>\r\n<hr><center>nginx/1.17.3</center>\r\n</body>\r\n</html>\r\n"
```

Reply:
Quay.io is known to be non-compliant, what else is new?

Reply:
That's news for me at least. I'll try with nginx pointed to another registry...

Reply:
I’m sorry, this was too flippant and not helpful.  If we are centralizing error handling, adding a centralized workaround for Quay (e.g. “body starts with `<html>`”) would of course be an option. (Sure, technically it was an option before as well, but sprinkling that workaround all over the package would be too ugly.)

Reply:
I left this unchanged. Improving and further centralizing error handling sounds like a good idea but is not the intention of this PR. The new function was added to avoid too much boilerplate.